### PR TITLE
test(vscode): use stub.resolves over stub.returns with Promise

### PR DESCRIPTION
**Pull request by alestiago**
## Status

**READY**

## Description

Using `stub.resolves(foo)` (since it causes the stub to return a Promise which resolves to the provided value) instead of `stub.returns(Promise(foo)`.

Changes:
- Replaces occurrences of `stub.return(Promise(foo))` with `stub.resolves(foo)`

## Additional context

- [Sinon stub documentation](https://sinonjs.org/releases/latest/stubs/)

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [X] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: extensions/vscode/src/test/suite/commands/create.test.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ suite("create command", () => {
 
   suite("file open dialog", () => {
     test("is shown when Uri is undefined", async () => {
-      vscodeStub.window.showOpenDialog.returns(Promise.resolve(undefined));
+      vscodeStub.window.showOpenDialog.resolves(undefined);
```
Comment (erickzanardo) [inline extensions/vscode/src/test/suite/commands/create.test.ts:48]: Pass undefined is a little bit redundant as it is implicit

```suggestion
      vscodeStub.window.showOpenDialog.resolves();
```

File: extensions/vscode/src/test/suite/commands/new-middleware.test.ts [hunk 2]
```diff
@@ -53,7 +53,7 @@ suite("new-middleware command", () => {
 
   suite("file open dialog", () => {
     test("is shown when Uri is undefined and fails to resolve a path from workspace", async () => {
-      vscodeStub.window.showOpenDialog.returns(Promise.resolve(undefined));
+      vscodeStub.window.showOpenDialog.resolves(undefined);
```
Comment (erickzanardo) [inline extensions/vscode/src/test/suite/commands/new-middleware.test.ts:56]:
```suggestion
      vscodeStub.window.showOpenDialog.resolves();
```

File: extensions/vscode/src/test/suite/commands/new-middleware.test.ts [hunk 3]
```diff
@@ -89,15 +89,15 @@ suite("new-middleware command", () => {
     const errorMessage = "Please select a valid directory";
 
     test("is shown when Uri is undefined and selected file is undefined", async () => {
-      vscodeStub.window.showOpenDialog.returns(Promise.resolve(undefined));
+      vscodeStub.window.showOpenDialog.resolves(undefined);
```
Comment (erickzanardo) [inline extensions/vscode/src/test/suite/commands/new-middleware.test.ts:92]:
```suggestion
      vscodeStub.window.showOpenDialog.resolves();
```

File: extensions/vscode/src/test/suite/commands/new-route.test.ts [hunk 4]
```diff
@@ -126,7 +126,7 @@ suite("new-route command", () => {
   suite("file open dialog", () => {
     test("is shown when Uri is undefined and fails to resolve a path from workspace", async () => {
       vscodeStub.window.showInputBox.returns(validRouteName);
-      vscodeStub.window.showOpenDialog.returns(Promise.resolve(undefined));
+      vscodeStub.window.showOpenDialog.resolves(undefined);
```
Comment (erickzanardo) [inline extensions/vscode/src/test/suite/commands/new-route.test.ts:129]:
```suggestion
      vscodeStub.window.showOpenDialog.resolves();
```

File: extensions/vscode/src/test/suite/commands/new-route.test.ts [hunk 5]
```diff
@@ -164,7 +164,7 @@ suite("new-route command", () => {
 
     test("is shown when Uri is undefined and selected file is undefined", async () => {
       vscodeStub.window.showInputBox.returns(validRouteName);
-      vscodeStub.window.showOpenDialog.returns(Promise.resolve(undefined));
+      vscodeStub.window.showOpenDialog.resolves(undefined);
```
Comment (erickzanardo) [inline extensions/vscode/src/test/suite/commands/new-route.test.ts:167]:
```suggestion
      vscodeStub.window.showOpenDialog.resolves();
```

**Review by erickzanardo — DISMISSED** [review]

**Review by wolfenrain — DISMISSED** [review]

**Pull request merged**